include/deal.II/base/patterns.h

```diff
@@ -2384,6 +2388,64 @@ namespace Patterns
       }
     };
 
+#ifdef DEAL_II_WITH_MAGIC_ENUM
+    // Enums
+    template <class T>
+    struct Convert<T, typename std::enable_if<std::is_enum<T>::value>::type>
+    {
+      static std::unique_ptr<Patterns::PatternBase>
+      to_pattern()
+      {
+        const auto               n     = magic_enum::enum_names<T>();
+        std::vector<std::string> names = {n.begin(), n.end()};
+        const auto               selection =
+          Patterns::Tools::Convert<decltype(names)>::to_string(
+            names,
+            Patterns::List(
+              Patterns::Anything(), names.size(), names.size(), "|"));
+        // Allow parsing a list of enums, and make bitwise or between them
+        return Patterns::List(Patterns::Selection(selection),
+                              0,
+                              names.size(),
+                              "|")
+          .clone();
+      }
+
+      static std::string
+      to_string(const T                     &value,
+                const Patterns::PatternBase &p = *Convert<T>::to_pattern())
+      {
+        namespace B                     = magic_enum::bitwise_operators;
```

Comment:
I find it a bit convoluted to introduce a namespace just to avoid rewriting magic_enum::bitwise_operators later on. I would avoid introducing this dummy namespace IMO. Especially since this is a 6 lines function.